APIKit/API.swift

```diff
@@ -1,142 +1,62 @@
     }
 
-    public class var acceptableStatusCodes: Set<Int> {
-        return Set(200..<300)
-    }
-
     private static let internalDefaultURLSession = NSURLSession(
         configuration: NSURLSessionConfiguration.defaultSessionConfiguration(),
         delegate: URLSessionDelegate(),
         delegateQueue: nil
     )
 
-    /// Creates a NSURLRequest instance from the specified HTTP method, path string
-    /// and parameters dictionary.
-    ///
-    /// Returns a mutable URL request instance which is meant to be modified in
-    /// subclasses or in `Request` protocol conforming types.
-    public class func URLRequest(#method: Method, path: String, parameters: [String: AnyObject] = [:], requestBodyBuilder: RequestBodyBuilder = requestBodyBuilder) -> NSMutableURLRequest? {
-        if let components = NSURLComponents(URL: baseURL, resolvingAgainstBaseURL: true) {
-            let request = NSMutableURLRequest()
-            
-            switch method {
-            case .GET, .HEAD, .DELETE:
-                components.query = URLEncodedSerialization.stringFromObject(parameters, encoding: NSUTF8StringEncoding)
-                
-            default:
-                switch requestBodyBuilder.buildBodyFromObject(parameters) {
-                case .Success(let box):
-                    request.HTTPBody = box.value
-                    
-                case .Failure(let box):
-                    return nil
-                }
-            }
-            
-            components.path = (components.path ?? "").stringByAppendingPathComponent(path)
-            request.URL = components.URL
-            request.HTTPMethod = method.rawValue
-            request.setValue(requestBodyBuilder.contentTypeHeader, forHTTPHeaderField: "Content-Type")
-            request.setValue(responseBodyParser.acceptHeader, forHTTPHeaderField: "Accept")
-            
-            return request
-        } else {
-            return nil
-        }
-    }
-    
-    @availability(*, unavailable, renamed="URLRequest(method:path:parameters:requestBodyBuilder)")
-    public class func URLRequest(method: Method, _ path: String, _ parameters: [String: AnyObject] = [:], requestBodyBuilder: RequestBodyBuilder = requestBodyBuilder) -> NSURLRequest? {
-        return URLRequest(method: method, path: path, parameters: parameters, requestBodyBuilder: requestBodyBuilder)
-    }
-
     // send request and build response object
-    public class func sendRequest<T: Request>(request: T, URLSession: NSURLSession = defaultURLSession, handler: (Result<T.Response, NSError>) -> Void = {r in}) -> NSURLSessionDataTask? {
-        let mainQueue = dispatch_get_main_queue()
-        
-        if let URLRequest = request.URLRequest {
-            let task = URLSession.dataTaskWithRequest(URLRequest)
-            
+    public class func sendRequest<T: Request>(request: T, URLSession: NSURLSession = defaultURLSession, handler: (Result<T.Response, APIError>) -> Void = {r in}) {
```

Comment:
Why returning `NSURLSettionTask?` is removed? I think this make it difficult a bit to just cancel the created request/task afterward.